source/hic_hal/freescale/k20dx/gpio.c

```diff
@@ -57,24 +57,43 @@ void gpio_init(void)
     PIN_nRESET_GPIO->PDDR &= ~PIN_nRESET;
     PIN_nRESET_PORT->PCR[PIN_nRESET_BIT] = PORT_PCR_MUX(1);
 
-    // Keep powered off in bootloader mode
-    // to prevent the target from effecting the state
-    // of the reset line / reset button
     if (!daplink_is_bootloader()) {
         // configure pin as GPIO
         PIN_POWER_EN_PORT->PCR[PIN_POWER_EN_BIT] = PORT_PCR_MUX(1);
-        // force always on logic 1
-        PIN_POWER_EN_GPIO->PDOR |= 1UL << PIN_POWER_EN_BIT;
-        PIN_POWER_EN_GPIO->PDDR |= 1UL << PIN_POWER_EN_BIT;
+        // set output to 0
+        PIN_POWER_EN_GPIO->PCOR = PIN_POWER_EN;
+        // switch gpio to output
+        PIN_POWER_EN_GPIO->PDDR |= PIN_POWER_EN;
     }
+}
+
+void gpio_handle_usb_connected(gpio_usb_connect_state_t state)
```

Comment:
Does the gpio driver need to know if USB is connected or if the USB connection timed out? If not, what about making this a function just for enabling and disabling board power, something like ```gpio_set_board_pwr()```? That would also make the code in main.c more explicit that it is enabling board power.

Reply:
I did it this way in case there were other use cases for the HIC to know the USB connection state.  I can change it if you prefer.

Reply:
Did you have any other potential use cases in mind? If not it might be best to limit the scope to just enabling power.

Reply:
Not at the moment, no. I'll go ahead and change it. Thanks for the feedback.